src/init.cpp

```diff
@@ -1337,14 +1337,16 @@ bool AppInit2(Config &config, thread_group &threadGroup)
     RegisterNodeSignals(GetNodeSignals());
 
     // sanitize comments per BIP-0014, format user agent and check total size
-    std::vector<string> uacomments;
+    std::vector<string> uacomments = {};
     for (string &cmt : mapMultiArgs["-uacomment"])
     {
         if (cmt != SanitizeString(cmt, SAFE_CHARS_UA_COMMENT))
             return InitError(strprintf(_("User Agent comment (%s) contains unsafe characters."), cmt));
         uacomments.push_back(SanitizeString(cmt, SAFE_CHARS_UA_COMMENT));
     }
-    strSubVersion = FormatSubVersion(CLIENT_NAME, CLIENT_VERSION, uacomments);
+
+    strSubVersion = FormatSubVersion(CLIENT_NAME, CLIENT_VERSION, BUComments);
+
     if (strSubVersion.size() > MAX_SUBVERSION_LENGTH)
```

Comment:
With truncation being done in FormatSubVersion now, this check can never fail.
This also represents a small loss in functionality because now with truncation the user will never receive feedback that their uacomment(s) as specified exceed the maximum length, they will just silently be clipped to fit.  I don’t know if it should still constitute a failure to start since the string length is variable at runtime if the user changes EB/AD settings, but a warning would still be good when clipping occurs.

Reply:
fixed